lazer/contracts/solana/programs/pyth-lazer-solana-contract/src/lib.rs

```diff
@@ -128,5 +257,24 @@ pub struct Update<'info> {
         bump,
         has_one = top_authority,
     )]
-    pub storage: Account<'info, Storage>,
+    pub storage: Account<'info, StorageV2>,
+}
+
+#[derive(Accounts)]
+pub struct VerifyMessage<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    #[account(
+        seeds = [STORAGE_SEED],
+        bump,
+        has_one = treasury
+    )]
+    pub storage: Account<'info, StorageV2>,
+    /// CHECK: this account doesn't need additional constraints.
+    pub treasury: AccountInfo<'info>,
```

Comment:
shouldn't you check it matches the treasury inside storage?

Reply:
it is checked with the `has_one` constraint in `storage` account.